src/snapred/backend/data/LocalDataService.py

```diff
@@ -749,7 +749,16 @@ def readGroupingFiles(self):
         # collect list of all files in folder that are applicable extensions
         groupingFiles = []
         for extension in extensions:
-            groupingFiles += self._findMatchingFileList(f"{groupingFolder}/*.{extension}", throws=False)
+            groupingFiles.extend(self._findMatchingFileList(f"{groupingFolder}/*.{extension}", throws=False))
         if len(groupingFiles) < 1:
             raise RuntimeError(f"No grouping files found in {groupingFolder} for extensions {extensions}")
-        return groupingFiles
+        focusGroups = {}
+        for file in groupingFiles:
+            focusGroups[file] = FocusGroup(
+                name=self.groupingSchemaFromPath(file),
+                definition=file,
+            )
+        return focusGroups
```

Comment:
this section should be its own method that consumes readGroupingFiles rather than hijacking its output

Reply:
Can do. This was just the minimal change I needed to touch the frontend as little as possible.

Reply:
Should be better now